### ISSUE TITLE
Remove "Associated Types" from views

### ISSUE THREAD

**Issue by marehr** [seqan/product_backlog#51]
### Description

Remove Associated types from view classes (not their tierator classes!):
```cpp
    /*!\name Associated types
     * \{
     */
    //!\brief The value type (equals the value_t with any references removed).
    using value_type = std::remove_reference_t<value_t>;
    //!\brief The reference_type.
    using reference  = value_type &;
    //!\brief The const reference type.
    using const_reference = value_type const &;
    //!\brief The size_type is void, because this range is never sized.
    using size_type = void;
    //!\brief The type to store the difference of two iterators.
    using difference_type = ptrdiff_t;
    //!\}
```
e.g. here: https://github.com/seqan/seqan3/blob/dc48fbfb64797d452ed31c1d081e1af14a5fa7a1/include/seqan3/range/views/take.hpp#L286-L301

Do not remove `using iterator` or `using const_iterator` for now.

In general: 
- If the type you remove is not used - great!
- Otherwise replace it with `some_iterator_name::value_type`, or other types respectively

You might encounter problems that the iterator defines the types by referring to the view, then the definitions from the view must be moved into the iterator.

### Motivation
Wiep encountered some problem when replacing `seqan3::reference<urng_t const>`, because there is no equivalent in `std::ranges::` (note that this is a transformation trait and therefore not the same as `std::ranges::range_reference_t`).

Replacing the whole expression with

```c++
using const_reference   = std::ranges::range_reference_t<urng_t const>;
```

does not compile with ranges from the ranges library.

In this case, I ask myself why we even define those member types (publicly)?

I think at one point we (or was it with hannes?) already talked about this, but I couldn't find the resolution.

---

2020-04-06 We (already) decided in an earlier discussion to remove those fields from `views` (not containers and not within iteraters)

**Comment by wvdtoorn** [comment]
@marehr This can be closed now, right?

**Comment by smehringer** [comment]
Yep

**Comment by smehringer** [comment]
Fixed by  seqan/seqan3#1755
